sway-ast/src/item/mod.rs

```diff
@@ -115,11 +115,22 @@ impl Spanned for FnSignature {
         };
         let end = match &self.where_clause_opt {
             Some(where_clause) => where_clause.span(),
-            None => match &self.return_type_opt {
-                Some((_right_arrow, ty)) => ty.span(),
-                None => self.arguments.span(),
+            None => match &self.return_type {
+                FnReturnType::TypedReturn((_right_arrow, ty)) => ty.span(),
+                FnReturnType::Implicit(_) => self.arguments.span(),
             },
         };
         Span::join(start, end)
     }
 }
+
+#[derive(Clone, Debug, Serialize)]
+pub enum FnReturnType {
+    Implicit(ImplicitReturn),
+    TypedReturn((RightArrowToken, Ty)),
+}
+
+#[derive(Clone, Debug, Serialize)]
+pub struct ImplicitReturn {
```

Comment:
Can we add a doc comment outining what this `Span` is pointing to when we have an implicit return type.